extensions/guacamole-auth-restrict/src/main/java/org/apache/guacamole/auth/restrict/RestrictionVerificationService.java

```diff
@@ -152,19 +149,42 @@ public static RestrictionType allowedByHostRestrictions(String allowedHostsStrin
             return RestrictionType.IMPLICIT_DENY;
         }
         
+        // Convert the string to a HostName
+        HostName remoteHostName = new HostName(remoteAddress);
+
+        // If for some reason we end up with a NULL value, here, something has gone wrong
+        // in attempting to convert the String form of the remote address into the
+        // HostName object, so we bail out and deny the action.
+        if (remoteHostName == null) {
+            LOGGER.error("Unable to parse remote address, action will be denied.");
+            LOGGER.debug("Error parsing \"{}\" into a valid hostname or IP address.", remoteAddress);
+            return RestrictionType.IMPLICIT_DENY;
+        }
+        
         // Split denied hosts attribute and process each entry, checking them
-        // against the current remote address, and returning false if a match is
-        // found.
+        // against the current remote address, and returning a deny restriction
+        // if a match is found, or if an error occurs in processing a host in
+        // the list.
         List<HostName> deniedHosts = HostRestrictionParser.parseHostList(deniedHostsString);
         for (HostName hostName : deniedHosts) {
+
+            if (hostName == null) {
+                LOGGER.warn("Host name is null, denying this action.");
+                LOGGER.debug("Received NULL host object, returning an implicit deny.");
+                return RestrictionType.IMPLICIT_DENY;
+            }
+
             try {
-                if (hostName.isAddress() && hostName.toAddress().contains(remoteHostName.asAddress()))
+                if (hostName.isAddress()
+                        && hostName.toAddress().contains(remoteHostName.asAddress())) {
                     return RestrictionType.EXPLICIT_DENY;
+                }
 
-                else
+                else {
                     for (IPAddress currAddr : hostName.toAllAddresses())
-                        if (currAddr.matches(remoteHostName.asAddressString()))
+                        if (currAddr != null && currAddr.matches(remoteHostName.asAddressString()))
```

Comment:
Checking the documentation of `toAllAddresses()`, it doesn't look like this function produces `null` results. It's documented as similar to `asAddress()`, except that it returns an empty array instead of `null`:

https://seancfoley.github.io/IPAddress/IPAddress/apidocs/inet/ipaddr/HostName.html#toAllAddresses--